### ISSUE TITLE
実装+実験: T1 Napoleon契約価値NNを競りruntimeへ接続し通しゲームで評価する

### ISSUE THREAD

**Issue by eletim** [eletim/napoleon-web#429]
Related: #202, #423, #427
Base branch: `feature/issue-202-non-playing-ai`

## 背景

#423で candidate=Napo / contract owner / target / suit固定のNapoleon契約価値NNを構築し、opening判定は大幅に改善した。

#427では、raise用teacherを

```text
candidate hand固定
→ remaining deck shuffle
→ RuleBased/Conservativeで競りを最初から実行
→ 実dealから自然発生したraise stateを収集
→ candidate=Napo・contract固定でteacher生成
```

へ変更した。

同じsingle-head architectureのまま、#423 checkpointをhistory-consistent raise datasetでfine-tuneしたT1がofflineで最良だった。

#421 raise diagnostic:
- B0 #423 M2: sign 67.5%, false PASS 12/40, EV bias -8.75
- T1: sign 80.0%, false PASS 7/40, EV bias -5.55

#422 opening diagnostic:
- T1: sign 88.9%, false PASS 0/45, false BID 5/45

境界付近の少数誤判定をこれ以上offlineで詰めるより、実際の競りruntimeで全体分布が自然かを評価する段階とする。

## 方針

T1を競りruntimeへ接続し、実ゲームを多数実行して以下を評価する。

- PASS/BID/raise率
- opening参加率
- raise率
- 最終Napoleon率
- 宣言target/suit分布
- contract success率
- relative reward
- hand strength別挙動
- illegal/fallback有無

このIssueではrewardやNN architectureを再設計しない。

## 使用model

#427 T1 checkpointを採用する。

```text
/tmp/issue427-t1-strong-raise-ft/checkpoint.pt
```

実装では `/tmp` 固定パスをruntime dependencyにしないこと。

既存のcheckpoint/config assetの配置規約を調査し、runtimeから再現可能に読み込める形へ組み込む。

モデル構造は#423 single-head M2:

```text
compact bidding input + selected action
→ MLP [512,512,256,256]
→ margin mean/std
→ Gaussian-derived pWin
```

### BID / raise候補のNapoleon EV

Napoleon raw reward:
- contract success: `+2d`
- contract failure: `0`

relative reward:

```text
R_N_win  = +7d/4
R_N_loss = -3d/4
```

したがって各合法BID/raise候補について:

```text
EV_N = pWin * (7d/4) + (1-pWin) * (-3d/4)
```

### Opening

current bidなしでは:

```text
PASS / All-Pass baseline = 0
```

合法BID候補の中で最大EVのactionを求め、

```text
max EV_N > 0 → そのBID
otherwise → PASS
```

を基本とする。

### Current bid exists

現時点の簡略化を維持する。

- BID/raise側: Napoleon EV
- PASS側: Citizen EV
- Adjutant / Napoleon-after-PASS / NoContractは今回再導入しない

Citizen側は既存#414/#420系の現行実装/モデルを再利用する。

既存runtimeにすでにPASS Citizen EV scorerがある場合はそれを使い、式やcheckpointを勝手に別物へ置き換えないこと。

各合法raise候補についてNapoleon EVを計算し、最大raise EVとPASS Citizen EVを比較する。

```text
max raise EV > PASS EV → raise
otherwise → PASS
```

## Action selection

重要:

単に「BIDするかPASSするか」だけでなく、全合法BID/raise候補をT1で評価し、EV最大の合法actionを選ぶ。

必ずlegalBidMaskに従う。

最低限ログ可能にする:
- selected action
- pWin
- predicted margin mean/std
- Napoleon EV
- PASS EV
- EV delta
- current bid context

通常runtimeで毎手大量ログを常時出す必要はない。diagnostic/simulation modeで集計できればよい。

## Runtime integration

既存の競りAI選択箇所を調査し、T1 scorerをproduction/runtime pathへ接続する。

要件:
- model load失敗時にillegal actionを出さない
- legal mask保証
- deterministic evaluation mode
- CPUで動作可能
- playing AI / exchange / adjutant selectionの既存挙動を壊さない

既存のRuleBased/Conservative fallbackを残す場合、そのfallback発生数を計測する。

## 比較baseline

最低限、同じseed panelで以下を比較する。

### A: 現行runtime bidding AI
親branch上で現在使われている競りAI。

### B: T1 runtime
今回のNapoleon EV + PASS Citizen EV scorer。

可能ならRuleBased/Conservative baselineも参考値として追加してよい。

## 通しゲーム評価

まずsmoke後、十分なゲーム数を実行する。

### smoke
- 20〜50 games
- illegal/fallback/crash確認

### main
最低1,000 games。
計算量が許せば5,000〜10,000 games。

A/Bは同じseed集合を使用してpaired比較可能にする。

playing policy、exchange、adjutant、opponent mix等は比較間で固定する。

### 競り全体
- total games
- all-pass / no-contract率
- bidding actions数/game
- PASS rate
- BID rate
- opening BID rate
- raise opportunity数
- raise rate given opportunity
- candidate最終Napoleon率

### 契約
- final target分布 13〜19
- final suit分布
- mean/median final target
- target別contract success率
- overall contract success率

### 手札強度別
既存strength classifierがあるなら:
- low
- medium
- strong
- veryStrong

別に:
- opening BID rate
- raise rate
- final Napoleon率
- contract success率

を出す。

### Reward
#420以降のrelative reward定義で:
- mean relative reward / player-game
- candidate視点mean relative reward
- role別 mean relative reward
- Napoleon時 mean relative reward
- Citizen時 mean relative reward

可能なら95% CIまたはbootstrap CIを付ける。

### Safety
- illegal bid count
- invalid mask count
- model load/inference failures
- fallback count
- game completion rate

illegal/fallbackが通常発生する設計でないなら0を目標とする。

## Decision quality diagnostic

通しゲーム中のdecisionをsampleし、最低限:
- openingでEVが0近傍のaction割合
- raiseでEV deltaが0近傍の割合
- 極端な低pWin BID
- 極端な高pWin PASS

を確認する。

目的は少数境界誤差ではなく、明らかな異常挙動がないかを見ること。

## 成功判定

T1 runtimeを採用候補とできる目安:

1. illegal/fallback/crashがない
2. opening/raise率が極端に0%/100%へcollapseしない
3. strong/veryStrongほどBID/raiseしやすい傾向が概ね出る
4. low handで無謀な高target契約が大量発生しない
5. contract success率が極端に低くない
6. relative rewardが現行runtime baselineより悪化しない、できれば改善
7. #427で改善したraise参加が実ゲームでも確認できる

単一の理想BID率を事前に固定しない。全体分布とrewardで判断する。

## Failure時

異常が出た場合は、即座にreward変更やarchitecture変更へ進まず、原因を分類する。

- pWin calibration
- PASS Citizen EV
- action ranking
- opening/raise distribution
- model runtime input mismatch

を切り分けて次Issue化する。

## 非目標

- reward再設計
- opening/raise Head分離
- 完全別NN
- Citizen/Adjutant再学習
- PPO再学習
- playing AI再学習
- exchange/adjutant AI再学習
- UI変更

## Tests

最低限:
1. T1 checkpoint load
2. runtime input schema compatibility
3. all legal BID candidates score可能
4. legalBidMask遵守
5. opening PASS threshold fixture
6. raise vs PASS Citizen EV fixture
7. max-EV legal action selection fixture
8. deterministic inference fixture
9. invalid model/fallback behavior
10. simulation smoke completion
11. no illegal bid fixture
12. relative reward aggregation fixture

## Git / PR

前提として #427 の必要変更/T1 artifactを親branchから利用可能な状態にすること。#427 PRが未mergeの場合は、勝手にmainへmergeせず、親branchへの取り込み状況を確認して必要な依存関係を整理する。

最新 `feature/issue-202-non-playing-ai` をbaseに専用branchを作成。

PR base:
`feature/issue-202-non-playing-ai`

main向けPRは禁止。

実装・simulation・評価・tests後、Codex review最大4回（review後修正最大3回）。blocking/actionable指摘がなくなればReadyにして親branchへmergeする。

mainにはmergeしない。

## 完了時に必ず報告

- runtime wiring箇所
- checkpoint配置/ロード方法
- exact decision式
- PASS Citizen EV source
- baseline AとT1 Bの設定
- game数/seeds
- PASS/BID/opening/raise率
- target/suit分布
- hand strength別挙動
- contract success
- relative reward比較
- illegal/fallback/crash
- decision boundary diagnostic
- T1を競りAIとして採用してよいか
- 次の問題がある場合その分類
- tests
- PR / merge commit / Codex review回数
- mainへmergeしていないこと
